### ISSUE TITLE
Failure when piping 200MB zip from blob storage to stream

### ISSUE THREAD

**Issue by winsome** [Azure/azure-storage-node#48]
I have some code that has worked in the past but has been having issues recently. 

There are files in blob storage that I'm expecting to be able to pipe to a stream. In production I'm piping to an http response, but in the example below I'm simply trying to pipe to a file. Both result in the same error.

    process.env.AZURE_STORAGE_ACCOUNT = 'storageAccount';
    process.env.AZURE_STORAGE_ACCESS_KEY = 'storageKey';    
    
    var require('fs');
    var require('azure-storage');
    
    function downloadBlob(containerName, fileName, stream)
    {
        var blobService = azure.createBlobService();
        blobService.getBlobProperties(containerName, fileName, function(error, properties, status){
            if(error || !status.isSuccessful)
            {
                console.log('Couldn\'t find file');
            }
            else
            {
                console.log('downloading file');
                var blobStream = blobService.createReadStream(containerName, fileName).pipe(stream);
            }; 
        }
    }

    var stream = fs.createWriteStream('download.zip');
    downloadBlob("container", "filename", stream);

Running this code results in the following error and I'm not sure what exactly has changed or where I need to go from here to fix it.
```
TypeError: Object function () {
        self.logger.debug('Write stream has ended');
        if (writeStream.close) {
          writeStream.close();
        }
        if (!savedBlobResult) {
          savedBlobResult = {};
        }
        savedBlobResult.contentMD5 = options.contentMD5;
        savedBlobResult.clientSideContentMD5 = null;
        if (md5Hash) {
          savedBlobResult.clientSideContentMD5 = md5Hash.digest('base64');
        }
        callback(error, savedBlobResult, savedBlobResponse);
      } has no method 'copy'
    at ChunkStream._copyToInternalBuffer (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\chunkstream.js:180:21)
    at ChunkStream._buildChunk (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\chunkstream.js:127:12)
    at ChunkStream.write (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\chunkstream.js:106:8)
    at ChunkStream.end (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\chunkstream.js:72:10)
    at EventEmitter.<anonymous> (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\services\blob\blobservice.js:4335:19)
    at EventEmitter.emit (events.js:98:17)    at BatchOperation._tryEmitEndEvent (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\batchoperation.js:279:19)
    at BatchOperation._fireOperationUserCallback (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\batchoperation.js:267:10)
    at BatchOperation._fireOperationUserCallback (C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\batchoperation.js:262:10)
    at C:\projects\aldis\GridSmartCloud\coreserver\node_modules\azure-storage\lib\common\streams\batchoperation.js:215:14
```

**Comment by yaxia** [comment]
@winsome Thanks for raising the issue. It’s a bug in the module that causes piping a large blob to stream to fail. We will fix this in an upcoming release.

**Comment by slepox** [comment]
This issue is fixed in latest release 0.4.3. Please check.
